WordPress/gutenberg
# Try/use featured image attribute

## What?
Continuation of #39410
Closes https://github.com/WordPress/gutenberg/issues/13795 and advances https://github.com/WordPress/gutenberg/issues/37753.


## Why?
We need ways to use featured image when doing page building, as well as site building, via core blocks.


## How?
Introduced a special prop that `BlockEdit` can use to set attributes based on contextual data it received from the editor. The idea is that the editor implementation (e.g. `edit-post`) has access to data and it needs to pass it to `BlockEdit` because this is the component that is always mounted in the editor, and we need it so that we can update blocks without hooking or modifying blocks themselves.

Changes to data that the editor implementation can access happen in various ways (from plugins, from the editor itself, from blocks) hence the need for things that don't appear and disappear, hence don't het unmounted.


## Testing Instructions
1. Switch to this PR
2. In a new post
3. Insert an image and/or a cover block
4. Set them to some image
5. From the inspector click on the top "Post" tab
6. Scroll down to Advanced and click on it
7. From the Set featured .. button set a featured image
8. Select the block(s) you added at (3)
9. In the Block toolbar next to Replace a new button (with a group icon for now) wills show
10. Click this button
11. The featured image should appear in the block11. 
12. Refresh the page
13. Without clicking on any block change the featured image
14. The change should be visible in the bound image block


## Screenshots or screencast 

https://user-images.githubusercontent.com/107534/158779381-2734f61a-6699-4dca-bda6-179d2150b55e.mp4
